init.lua

```diff
@@ -26,49 +26,51 @@ vim.g.loaded_zipPlugin = false
 vim.g.loaded_2html_plugin = false
 vim.g.loaded_remote_plugins = false
 
----- Doom Configurations ------------------------
--------------------------------------------------
-vim.defer_fn(function()
-  local load_modules = require("doom.utils").load_modules
+local doom_root, sep = require("doom.core.system").doom_root, require("doom.core.system").sep
+local compiled_plugins_path = ("%s%splugin%spacker_compiled.lua"):format(doom_root, sep, sep)
+local compiled_plugins_exists = require("doom.utils").file_exists(compiled_plugins_path)
+local load_modules = require("doom.utils").load_modules
+
+if not compiled_plugins_exists then
+  -- bootstrap start only -- no deferred loading
+  -- ensure packer availability only
+  load_modules("doom", {"modules"})
+  -- followings are blocked until packages are loaded
+  vim.cmd("PackerInstall")
+  vim.cmd("PackerCompile")
+  -- packages in start path may cause noise.  Ignore and type ":"
+  -- Once successful, use ":q!" to get out and restart nvim.
+else
+  ---- Doom Configurations ------------------------
+  -------------------------------------------------
+  vim.defer_fn(function()
 
-  -- Load Doom stuff (core, modules, extras)
-  load_modules("doom", { "core", "modules", "extras.autocmds" })
+    -- Load Doom stuff (core, modules, extras)
+    load_modules("doom", { "core", "modules", "extras.autocmds" })
 
-  -- If the dashboard plugin is already installed and the packer_compiled.lua
-  -- file exists so we can make sure that the dashboard have been loaded.
-  local doom_root, sep = require("doom.core.system").doom_root, require("doom.core.system").sep
-  local compiled_plugins_path = string.format(
-    "%s%splugin%spacker_compiled.lua",
-    doom_root,
-    sep,
-    sep
-  )
-  if require("doom.utils").file_exists(compiled_plugins_path) then
     -- If the current buffer name is empty then trigger Dashboard.
     -- NOTE: this is done to avoid some weird issues with Dashboard and
     --       number / signcolumn when Dashboard gets triggered automatically
     if (vim.api.nvim_buf_get_name(0):len() == 0) and packer_plugins["dashboard-nvim"] then
       vim.cmd("Dashboard")
     end
-  end
 
-  vim.opt.shadafile = ""
-  vim.cmd([[
-    rshada!
-    doautocmd BufRead
-    syntax on
-    filetype on
-    filetype plugin indent on
-    PackerLoad nvim-treesitter
-  ]])
+    vim.opt.shadafile = ""
+    vim.cmd([[
+      rshada!
+      doautocmd BufRead
+      syntax on
+      filetype on
+      filetype plugin indent on
+      PackerLoad nvim-treesitter
+    ]])
 
-  -- Load keybindings module at the end because the keybindings module cost is high
-  vim.defer_fn(function()
-    load_modules("doom.extras", { "keybindings" })
-    if not require("doom.core.functions").is_plugin_disabled("which-key") then
```

Comment:
I think this line should be still here to avoid issues when the user doesn't have which-key plugin installed

Reply:
You are absolutely correct.  When I rebased to develop branch, I must have broken it.  Excuse me.

Reply:
No worries, thank you for fixing it!